src/plugins/vis_types/timeseries/public/trigger_action/index.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+import { PaletteOutput } from '../../../../charts/public';
+import type {
+  NavigateToLensContext,
+  VisualizeEditorLayersContext,
+} from '../../../../visualizations/public';
+import type { Panel } from '../../common/types';
+import { PANEL_TYPES } from '../../common/enums';
+import { getDataSourceInfo } from './get_datasource_info';
+import { getSeries } from './get_series';
+import { getYExtents } from './get_extents';
+
+const SUPPORTED_FORMATTERS = ['bytes', 'percent', 'number'];
+
+/*
+ * This function is used to convert the TSVB model to compatible Lens model.
+ * Returns the Lens model, only if it is supported. If not, it returns null.
+ * In case of null, the menu item is disabled and the user can't navigate to Lens.
+ */
+export const triggerTSVBtoLensConfiguration = async (
+  model: Panel
+): Promise<NavigateToLensContext | null> => {
+  // Disables the option for not timeseries charts and for the string mode
+  if (model.type !== PANEL_TYPES.TIMESERIES || !model.use_kibana_indexes) {
+    return null;
+  }
+  const layersConfiguration: { [key: string]: VisualizeEditorLayersContext } = {};
+
+  // handle multiple layers/series
+  for (let layerIdx = 0; layerIdx < model.series.length; layerIdx++) {
+    const layer = model.series[layerIdx];
+    if (layer.hidden) continue;
+
+    const { indexPatternId, timeField } = await getDataSourceInfo(
```

Comment:
Do we have any metric about how many layers are out there for a TSVB visualization (on avg, max)?

Reply:
No unfortunately we don't.